### DISCUSSION TITLE
Configuring Geta.Optimizely.Categories 'Select Categories' tree

### DISCUSSION THREAD

**Discussion by geonetric-matt** [community/community#120908] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Greetings Brian & All:

I'm new to Geta.Optimizely.Categories and to Optimizely CMS and I have built a Categories prototype with using the Alloy demo site. Everything seems to be working well and I can control my categories in the way that's needed except for one case...

In my 'Select Categories' tree, I need to only display the parent-child category types for a given category and only child types should be selectable(allowed). I've tried the following combinations of appsettings and AllowedType attribution but, at best, have only been able to meet 2 of my 3 goals.	

I'm hoping that you can provide some guidance about whether all 3 goals can be achieved and, if so, how to go about achieving it.

Regards,
Matt

**Goals:**
	
In Select Categories tree:	
1. Display ProductCat1(parent) as RestrictedType
2. Display ProductChildCat1 & ProductChildCat2 as AllowedTypes
3. Do not display any other parent types such as NewsCat1 and SearchCat1

![image](https://github.com/community/community/assets/167918924/9b8284ab-c989-4182-86d2-a0f68c77806a)

**Try 1:**

Config:
- AllowedTypes: ProductCat1, ProductChildCat1 & ProductChildCat2
- HideDisallowedRootCategories = true

					
Result: In Select Categories tree
1. Fail-ProductCat1(parent) displays as an AllowedType
2. Pass-ProductChildCat1 & ProductChildCat2 display as AllowedTypes
3. Pass-Do not display any other parent types such as NewsCat1 and SearchCat1

![image](https://github.com/community/community/assets/167918924/58d8ae13-3a7d-419c-8c88-025e9fa69c46)

**Try 2:** 

Config:		
- AllowedTypes: ProductChildCat1 & ProductChildCat2
- RestrictedTypes: ProductCat1
- HideDisallowedRootCategories = true	

Result: In Select Categories tree - 
1. Fail-ProductCat1(parent) does not display.
2. Fail-ProductChildCat1 & ProductChildCat2 do not display.
3. Pass-Do not display any other parent types such as NewsCat1 and SearchCat1

![image](https://github.com/community/community/assets/167918924/68cf1a91-f318-4a0c-87c3-fc945d58f2b2)

**Try 3:**

Config:				
- AllowedTypes: ProductChildCat1 & ProductChildCat2
- RestrictedTypes: ProductCat1
- HideDisallowedRootCategories = false			

Result: In Select Categories tree -
1. Pass-Display ProductCat1(parent) as RestrictedType
2. Pass-Display ProductChildCat1 & ProductChildCat2 as AllowedTypes
3. Fail-Do not display any other parent types such as NewsCat1 and SearchCat1

![image](https://github.com/community/community/assets/167918924/55f8c7ab-df2b-4309-877b-d859e1b01c2c)

**Comment by Zesaa** [comment]
thanks

**Comment by ebndev** [comment]
Hi @geonetric-matt,

Thanks for being a part of the GitHub Community, we're glad you're here!

If you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on support for Alloy. 

The GitHub Community Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can and reaching the people you intended this message to go to, but this space may not be the right place for this particular topic.

Best of luck!

**Comment by geonetric-matt** [comment]
I fail to see how this doesn't apply to the GitHub community.

This question is related to Geta.Optimizely.Categories, which is a GitHub repo: https://github.com/Geta/geta-optimizely-categories/ 

Does this not qualify as a GitHub Community Discussion?

**Reply by ebndev** [reply]
Hey @geonetric-matt, your posts are definitely welcomed here!

I saw that you addressed your post with "Greetings Brian & All:" and it looked to me like you were trying to reach out to the developers in that specific project, which often happens in Community Discussions. I wrote my previous comment with the intention that you could put your message in a space where the people you're trying to reach may be more likely to see it and provide you with help, such as opening an issue on their repository because this is a public forum.

Let me know if you'd like me to reopen your discussion if you were looking for help from other users!
